agent/xds/endpoints.go

```diff
@@ -353,6 +360,37 @@ func (s *ResourceGenerator) endpointsFromServicesAndResolvers(
 	return resources, nil
 }
 
+func (s *ResourceGenerator) makeEndpointsForOutgoingPeeredServices(
+	cfgSnap *proxycfg.ConfigSnapshot,
+) ([]proto.Message, error) {
+	var resources []proto.Message
+
+	// generate the endpoints for the linked service groups
+	for _, serviceGroups := range cfgSnap.MeshGateway.PeeringServices {
+		for sn, serviceGroup := range serviceGroups {
+			var clusterName string
+			if len(serviceGroup) > 0 {
```

Comment:
Doesn't seem like we should proceed if `len(serviceGroup) == 0`, since there won't be any endpoints and the clusterName won't be populated